core/server/src/main/java/alluxio/worker/file/FileSystemWorker.java

```diff
@@ -128,13 +144,54 @@ public long createUfsFile(AlluxioURI ufsUri) throws FileAlreadyExistsException,
   }
 
   /**
+   * Opens a stream to the under file system file denoted by the temporary file id. This call
+   * will skip to the position specified in the file before returning the stream. The caller of
+   * this method is required to close the stream after they have finished operations on it.
+   *
+   * @param tempUfsFileId the worker specific temporary file id for the file in the under storage
+   * @param position the absolute position in the file to position the stream at before returning
+   * @return an input stream to the ufs file positioned at the given position
+   * @throws FileDoesNotExistException if the worker file id is invalid
+   * @throws IOException if an error occurs interacting with the under file system
+   */
+  public InputStream getUfsInputStream(long tempUfsFileId, long position)
+      throws FileDoesNotExistException, IOException {
+    return mUnderFileSystemManager.getInputStreamAtPosition(tempUfsFileId, position);
+  }
+
+  /**
+   * Returns the output stream to the under file system file denoted by the temporary file id.
+   * The stream should not be closed by the caller.
+   *
+   * @param tempUfsFileId the worker specific temporary file id for the file in the under storage
+   * @return the output stream writing the contents of the file
+   * @throws FileDoesNotExistException if the temporary file id is invalid
+   */
+  public OutputStream getUfsOutputStream(long tempUfsFileId) throws FileDoesNotExistException {
+    return mUnderFileSystemManager.getOutputStream(tempUfsFileId);
+  }
+
+  /**
    * @return the worker service handler
    */
   public FileSystemWorkerClientServiceHandler getWorkerServiceHandler() {
     return mServiceHandler;
   }
 
   /**
+   * Opens a file in the under file system and registers a temporary file id with the file. This
+   * id is valid until the file is closed.
```

Comment:
Yes, both for reads and writes. However for the write, the first writer will win. Also, it is up to the ufs implementation to ensure there is a lock on the file when it is being read.